keras/src/utils/jax_layer.py

```diff
@@ -342,8 +442,15 @@ def _get_init_rng(self):
         Returns:
             a JAX `PRNGKey` or structure of `PRNGKey`s that will be passed as
             the `rng` argument of `init_fn`.
+
         """
-        return self.seed_generator.next()
+
+        from keras.src.backend.jax.core import convert_to_tensor
+
+        if backend.backend() == "jax":
+            return self.seed_generator.next()
+        if backend.backend() == "tensorflow":
+            return convert_to_tensor(self.seed_generator.next())
```

Comment:
Actually undo this and do `return self.seed_generator.next()`. It's how it's documented right now.

Let's do the conversion where it's used to make it easier to override this (see my other comment).